### PR TITLE
Add release notes for 2.17

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Add release notes for 2.17 release
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Maintenance
* Why these changes are required?
Release 2.17
* What is the old behavior before changes and new behavior after changes?
Nothing

### Issues Resolved
None
Is this a backport? If so, please add backport PR # and/or commits #, and remove `backport-failed` label from the original PR.
No

Do these changes introduce new permission(s) to be displayed in the static dropdown on the front-end? If so, please open a draft PR in the security dashboards plugin and link the draft PR here
No
### Testing
None
### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] New Roles/Permissions have a corresponding security dashboards plugin PR
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-security.release-notes-2.17.0.0.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+## Version 2.17.0 Release Notes
+
+Compatible with OpenSearch and OpenSearch Dashboards version 2.17.0
+
+### Enhancements
+* Add `ignore_hosts` config option for auth failure listener ([#4538](https://github.com/opensearch-project/security/pull/4538))
+* added API roles for correlationAlerts ([#4689](https://github.com/opensearch-project/security/pull/4689))
+* Manually backport #4632 (Allow multiple signing keys to be provided) ([#4666](https://github.com/opensearch-project/security/pull/4666))
```
Comment (cwperks) [inline release-notes/opensearch-security.release-notes-2.17.0.0.md:8]:
```suggestion
* Allow multiple signing keys to be provided ([#4666](https://github.com/opensearch-project/security/pull/4666))
```

File: release-notes/opensearch-security.release-notes-2.17.0.0.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* Permission changes for correlationAlerts ([#4704](https://github.com/opensearch-project/security/pull/4704))
+
+### Bug Fixes
+* Addresses a bug with `plugins.security.allow_unsafe_democertificates` setting ([#4603](https://github.com/opensearch-project/security/pull/4603))
+* Fix covereage-report workflow ([#4684](https://github.com/opensearch-project/security/pull/4684), [#4683](https://github.com/opensearch-project/security/pull/4683))
+* Handle the audit config being null ([#4664](https://github.com/opensearch-project/security/pull/4664))
+* Fixes authtoken endpoint ([#4631](https://github.com/opensearch-project/security/pull/4631))
+* Fixed READ_ACTIONS required by TermsAggregationEvaluator ([#4607](https://github.com/opensearch-project/security/pull/4607))
+* Sort the DNS Names in the SANs ([#4640](https://github.com/opensearch-project/security/pull/4640))
+
```
Comment (cwperks) [inline release-notes/opensearch-security.release-notes-2.17.0.0.md:19]: nit: remove the extra new lines

**Review by cwperks — APPROVED** [review]

File: release-notes/opensearch-security.release-notes-2.17.0.0.md (hunk 2 above)
Comment (derek-ho) [reply release-notes/opensearch-security.release-notes-2.17.0.0.md:19]: Done in: [4ebe0ed](https://github.com/opensearch-project/security/pull/4705/commits/4ebe0ed29bb91ae02dedb8779d87fbedd53b782f)

File: release-notes/opensearch-security.release-notes-2.17.0.0.md (hunk 1 above)
Comment (derek-ho) [reply release-notes/opensearch-security.release-notes-2.17.0.0.md]: Done in: [4ebe0ed](https://github.com/opensearch-project/security/pull/4705/commits/4ebe0ed29bb91ae02dedb8779d87fbedd53b782f)

**Review by RyanL1997 — APPROVED** [review]

**Pull request merged**